### PR TITLE
External plugins load after confirming config warning

### LINKED ISSUE #1692: plugins from external sessions should not load until external-session warning is confirmed
Non-core plugins should not load until the "can contain code that runs" warning is clicked through.

To reproduce:

1. open the dev tools network panel
2. go to https://s3.amazonaws.com/jbrowse.org/code/jb2/master/index.html?config=https%3A%2F%2Funpkg.com%2Fjbrowse-plugin-msaview%2Fdist%2Fconfig.json&session=share-I0uAH935oY&password=b6Jvm

you will see the msa view plugin loading, but it should not load until the warning is confirmed

### PR DESCRIPTION
**Pull request by peterkxie**
This closes #1692, now the config warning modal holds off fetchPlugins until after the user confirms that they trust the source, so external plugins get loaded after. Cancel resets to default configuration (same logic as onFactoryReset)

Making as draft for now because not too sure how to handle the cancel case. Assuming they do not want to load the cross origin config, will have to atleast clear the config in the URL and in the configPath and set it to the default config (config.json). If clearing the config, may have to clear the session since that session may have info recorded using the previous config (such as confirming, loading a view, and then refresh and not confirming/cancel the second time).

### CODE REVIEW

**Comment by cmdcolin** [comment]
Visiting http://localhost:3000/?config=test_data/config_demo.json has an error currently about GDCTrack not being found on this branch

File: products/jbrowse-web/src/configWarningModal.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              await loader.fetchPlugins(
+                session as { plugins: PluginDefinition[] },
+              )
+              loader.setConfigSnapshot({
+                ...session,
+                id: shortid(),
+              })
+              handleClose()
+            }}
+          >
```
Comment (rbuels) [inline products/jbrowse-web/src/configWarningModal.tsx:82]: this fetching code should not be in the component, you should probably pass a `onConfirm` callback into this component

File: products/jbrowse-web/src/sessionWarningModal.tsx [hunk 2]
```diff
@@ -54,31 +54,27 @@ export default function SessionWarningModal({
...
-                : loader.setConfigSnapshot({
-                    ...session,
-                    id: shortid(),
-                  })
+            onClick={async () => {
+              loader.setSessionSnapshot({
+                ...session,
+                id: shortid(),
+              })
               handleClose()
```
Comment (rbuels) [inline products/jbrowse-web/src/sessionWarningModal.tsx:74]: Likewise, should probably pass`onConfirm` and `onCancel` callbacks to this component

**Comment by cmdcolin** [comment]
Is this one ready now?

**Comment by cmdcolin** [comment]
Looks good to me :)

**Comment by cmdcolin** [comment]
merge'd

**Pull request merged**